addons/sale_timesheet/views/project_update_templates.xml

```diff
@@ -31,17 +32,80 @@
 <div name="profitability" t-if="show_profitability">
 <t t-if="project.analytic_account_id and project.allow_billable and user.has_group('project.group_project_manager')" name="costs">
 <h3 style="font-weight: bolder"><u>Profitability</u></h3>
-<t t-if="project.analytic_account_id.line_ids"> The cost of the project is now at <t t-esc="profitability['costs_formatted']"/>, for a revenue of <t t-esc="profitability['revenues_formatted']"/>, leading to a
+<t t-if="project.analytic_account_id.line_ids"> The cost of the project is now at <t t-out="profitability['costs_formatted']['total_cost']"/>, for a revenue of <t t-out="profitability['revenues_formatted']['total_revenue']"/>, leading to a
 <span>
 <font t-if="profitability['margin'] &gt; 0"  style="color: rgb(0, 128, 0)">
-<b><t t-esc="profitability['margin_formatted']"/></b>
+<b><t t-out="profitability['margin_formatted']"/></b>
 </font>
 <font t-elif="profitability['margin'] &lt; 0" style="color: rgb(128, 0, 0)">
-<b><t t-esc="profitability['margin_formatted']"/></b>
+<b><t t-out="profitability['margin_formatted']"/></b>
 </font>
-<t t-else="" t-esc="profitability['margin_formatted']"/>
-</span> margin (<t t-esc="profitability['margin_percentage']"/>%).
+<t t-else="" t-out="profitability['margin_formatted']"/>
+</span> margin (<t t-out="profitability['margin_percentage']"/>%).
 </t>
+
+<div t-if="profitability['profitability_items']" name="profitability_detail" class="mt-4">
+    <table class="table table-striped">
+        <thead class="border-2 border-start-0 border-end-0">
+            <tr>
+                <th class="fw-bolder">Revenues</th>
+                <th class="fw-bolder text-end">Invoiced</th>
+                <th class="fw-bolder text-end">To Invoice</th>
+                <th class="fw-bolder text-end">Expected</th>
+            </tr>
+        </thead>
+        <tbody>
+            <t t-set="revenues" t-value="profitability['profitability_items']['revenues']"/>
+            <tr t-foreach="revenues['data']" t-as="revenue">
+                <td t-out="profitability_labels[revenue['id']]"/>
+                <td class="text-end" t-out="revenue['formatted']['invoiced']"/>
+                <td class="text-end" t-out="revenue['formatted']['to_invoice']"/>
+                <td class="text-end" t-out="revenue['formatted']['expected_revenue']"/>
+            </tr>
+            <tfoot>
+                <td class="fw-bolder text-end">Total</td>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_invoiced']"/>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_to_invoice']"/>
+                <td class="fw-bolder text-end" t-out="profitability['revenues_formatted']['total_revenue']"/>
```

Comment:
IMO, `profitability` data can be replaced by `costs_formatted` or `revenues_formatted`, I don't really understand why we have to import it to use it in that template since we normally have all the data needed. 🤔

Reply:
> IMO, `profitability` data can be replaced by `costs_formatted` or `revenues_formatted`, I don't understand why we have to import it to use it in that template since we normally have all the data needed. 🤔

1) I split the costs_formatted and revenues_formatted inside the profitability.
2) Amount data is not formatted so we have to format the amount on server side first and then we import it.

thanks